### PR TITLE
docs: ringing docs improvements

### PR DESCRIPTION
**Pull request by Brazol**

### CODE REVIEW

File: docusaurus/docs/Flutter/05-advanced/02-ringing.mdx [hunk 1]
```diff
@@ -40,24 +40,28 @@ This JSON file contains the credentials that need to be uploaded to Stream’s s
...
+- Add your previously generated Firebase Credentials in the **Credentials JSON** field.
+
+- Enable this provider using toggle button
 
 - Click **Create** and your push provider should be ready.
 
 #### Step 3 - Add dependencies to your app
 
-To integrate push notifications in your Flutter app, you need to use the package [`firebase_messaging`](https://pub.dev/packages/firebase_messaging).
+To integrate push notifications in your Flutter app, you need to use the **firebase_messaging** package.
```
[the remark this answers is not in the document]
Comment (Brazol) [reply docusaurus/docs/Flutter/05-advanced/02-ringing.mdx]: Good question :) I think I wanted to use it somewhere below but didn't. I'm putting it back in

File: docusaurus/docs/Flutter/05-advanced/02-ringing.mdx [hunk 2]
```diff
@@ -40,24 +40,28 @@ This JSON file contains the credentials that need to be uploaded to Stream’s s
...
+- Go to the dashboard of your video project at the [Stream website](https://dashboard.getstream.io).
 
 - Open the **Push Notifications** tab under **Video & Audio**.
 
 - Select **New Configuration** and select **Firebase**.
 
 ![Firebase Configuration](../assets/advanced_assets/firebase_config.png)
 
-- Add a name for your push provider in the **Name** field and add your previously generated Firebase Credentials in the **Credentials JSON** field.
+- Add a name for your push provider in the **Name** field. You will use this name later in the code to identify which provider to use for android notifications.
```
Comment (deven98) [inline docusaurus/docs/Flutter/05-advanced/02-ringing.mdx:51]:
```suggestion
- Add a name for your push provider in the **Name** field. You will use this name later in the code to identify which provider to use for Android notifications.
```

File: docusaurus/docs/Flutter/05-advanced/02-ringing.mdx [hunk 3]
```diff
@@ -40,24 +40,28 @@ This JSON file contains the credentials that need to be uploaded to Stream’s s
...
+
+- Add your previously generated Firebase Credentials in the **Credentials JSON** field.
+
+- Enable this provider using toggle button
```
Comment (deven98) [inline docusaurus/docs/Flutter/05-advanced/02-ringing.mdx:55]:
```suggestion
- Enable this provider using toggle button.
```

File: docusaurus/docs/Flutter/05-advanced/02-ringing.mdx [hunk 4]
```diff
@@ -132,11 +140,138 @@ Future<void> _handleRemoteMessage(RemoteMessage message) async {
...
+}
+
+_observeFcmMessages() {
+  FirebaseMessaging.onBackgroundMessage(_firebaseMessagingBackgroundHandler);
+  _fcmSubscription = FirebaseMessaging.onMessage.listen(_handleRemoteMessage);
+}
+```
+
 In a high-level widget in your app, add this code to listen to CallKit events:
+- remember to implement navigation in a marked line
```
Comment (deven98) [inline docusaurus/docs/Flutter/05-advanced/02-ringing.mdx:264]: This line is a bit unclear. Can we rephrase or add it later as a note?

File: docusaurus/docs/Flutter/05-advanced/02-ringing.mdx [hunk 5]
```diff
@@ -132,11 +140,138 @@ Future<void> _handleRemoteMessage(RemoteMessage message) async {
...
+
+- Go to the dashboard of your video project at the [Stream website](https://dashboard.getstream.io).
+
+- Open the **Push Notifications** tab under **Video & Audio**.
+
+- Select **New Configuration** and select **APN**.
+
+![APNs Configuration](../assets/advanced_assets/apns_config.png)
+
+- Add a name for your push provider in the **Name** field. This is the name used while setting up ios push notifications in the code we did [earlier](#Step-5---Add-code-to-listen-to-push-notifications)
```
Comment (deven98) [inline docusaurus/docs/Flutter/05-advanced/02-ringing.mdx:209]:
```suggestion
- Add a name for your push provider in the **Name** field. This is the name used while setting up iOS push notifications in the code we did [earlier](#Step-5---Add-code-to-listen-to-push-notifications)
```

File: docusaurus/docs/Flutter/05-advanced/02-ringing.mdx [hunk 6]
```diff
@@ -132,11 +140,137 @@ Future<void> _handleRemoteMessage(RemoteMessage message) async {
...
+  final call = await StreamVideo.instance.consumeIncomingCall(
+    uuid: calls.first.uuid,
+    cid: calls.first.callCid,
+  );
+}
+```
+
+#### Step 6 - Request notification permission from user
+
+For Android 13+ you need to request post notification permission. You can do it using [permission_handler](https://pub.dev/packages/permission_handler) package.
```
Comment (deven98) [inline docusaurus/docs/Flutter/05-advanced/02-ringing.mdx:182]:
```suggestion
For Android 13+ you need to request the `POST_NOTIFICATIONS` permission. You can do it using the [permission_handler](https://pub.dev/packages/permission_handler) package.
```

File: docusaurus/docs/Flutter/05-advanced/02-ringing.mdx [hunk 7]
```diff
@@ -132,11 +140,137 @@ Future<void> _handleRemoteMessage(RemoteMessage message) async {
...
+```plist
+<key>NSCameraUsageDescription</key>
+<string>$(PRODUCT_NAME) Camera Usage!</string>
+<key>NSMicrophoneUsageDescription</key>
+<string>$(PRODUCT_NAME) Microphone Usage!</string>
+<key>UIApplicationSupportsIndirectInputEvents</key>
+<true/>
+<key>BGTaskSchedulerPermittedIdentifiers</key>
+<array>
+  <string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
```
Comment (deven98) [inline docusaurus/docs/Flutter/05-advanced/02-ringing.mdx:234]:
```suggestion
	 <string>$(PRODUCT_BUNDLE_IDENTIFIER)</string>
```

**Review by deven98 — APPROVED** [review]
LGTM

**Pull request merged**